src/test/clojure/xtdb/metrics_test.clj

```diff
@@ -22,3 +24,29 @@
     (t/is (= 4.0 (.count ^Counter (.counter (.find registry "query.error")))))
 
     (t/is (= 2.0 (.count ^Counter (.counter (.find registry "query.warning")))))))
+
+(t/deftest test-transaction-exception-counter
+  (let [registry ^CompositeMeterRegistry (tu/component tu/*node* :xtdb.metrics/registry)]
+    (t/is (thrown-with-msg? Exception #"" (jdbc/execute! tu/*conn* ["INSERT INTO foo (a) VALUES (42)"])))
+
+    (t/testing "DML enabled with BEGIN READ WRITE"
+      (jdbc/execute! tu/*conn* ["START TRANSACTION"])
+      (jdbc/execute! tu/*conn* ["INSERT INTO foo (a) VALUES (42)"])
+      (t/is (thrown-with-msg? Exception #"" (jdbc/execute! tu/*conn* ["COMMIT"]))))
+
+    (t/is (= 0.0 (.count ^Counter (.counter (.find registry "query.error")))))
+    (t/is (= 0.0 (.count ^Counter (.counter (.find registry "query.warning")))))
+    (t/is (= 2.0 (.count ^Counter (.counter (.find registry "tx.error")))))))
```

Comment:
What happens here if you add something that is an async error in the indexer in the above test going through jdbc. Will it get counted twice? I think this will show you if the whole approach is sound.

Reply:
I'm not honestly sure how to test that. Is there something we can pass in that parses correctly but causes an error in the indexer?

Reply:
Try something like `INSERT INTO docs SELECT 1/0 AS _id`

Reply:
Yeah this does double count, I'm not really sure how to prevent it

Reply:
I've added another condition (not ideal I know) for a :dml-result error type so that we can do nothing in that case where we know the indexer will have already counted the error.